### PR TITLE
Combined functionality

### PR DESCRIPTION
**Pull request by mkerr6872**

### CODE REVIEW

File: project/Scripts/UnitTests.gd [hunk 1]
```diff
@@ -0,0 +1,95 @@
+extends Node
+
+#Test addNode
+func test_addNode():
+	var graph = ProofGraph.new()
+	for i in range(0, 50, 1):
+		var nodeInfo = {"location": [i, i+1, i+2], "node_id": i}
+		graph.addNode(nodeInfo)
```
Comment (James-Oswald) [inline project/Scripts/UnitTests.gd:8]: These tests aren't robust enough.

There is nothing special about 2-50 that isn't already being tested by 1.
Unit tests should focus on all types of inputs and edge cases: weird situations where the code may break, but needs to be resilient.

This should minimally test 
1) negative locations
2) floating point locations
3) massive locations (around the integer/float limit)
4) illegal locations: IE: Over the integer limit, Special floating values: NaN, Inf, -Inf. and fail gracefully. 



and show it is reliant against them.

File: project/Scripts/UnitTests.gd
Comment (James-Oswald) [inline project/Scripts/UnitTests.gd:1]: The comments I leave on the first test generalize to all the tests.

File: extension/proofgraph.h [hunk 2]
```diff
@@ -12,31 +12,36 @@
...
+#include "godot_cpp/classes/mesh_instance3d.hpp"
+#include "godot_cpp/classes/mesh.hpp"
+#include "godot_cpp/classes/box_mesh.hpp"
+#include "godot_cpp/classes/text_mesh.hpp"
+#include "godot_cpp/classes/standard_material3d.hpp"
+#include "godot_cpp/variant/color.hpp"
+#include "godot_cpp/classes/cylinder_mesh.hpp"
+#include "godot_cpp/classes/box_shape3d.hpp"
+#include "godot_cpp/classes/collision_shape3d.hpp"
+#include "godot_cpp/classes/static_body3d.hpp"
```
Comment (James-Oswald) [inline extension/proofgraph.h:24]: Why are we including all of these in the H file if we're not using any of them in the H file? Should be imported in the cpp file where they are actually used.

**Review by James-Oswald — APPROVED** [review]
